src/renderer/actions/repair.js

```diff
@@ -14,76 +13,128 @@ import kill from "./utils/kill";
+
         try {
-            const versionDir = (await fs.readdir(roaming)).find(d => d.split(".").length > 2);
-            const modulesPath = path.join(roaming, versionDir, "modules");
-            log("Removing " + modulesPath);
+            // Detect the latest Discord version directory in this path.
+            const version = findLatestDiscordVersion(discordConfigPath);
+            if (!version) continue;  // Skip path since nothing was found.
+
+            // Cross-platform: Remove the "modules" directory for that version.
+            // NOTE: This will cause Discord to redownload the latest version again on next launch.
+            const modulesPath = path.join(discordConfigPath, version, "modules");
             if (await exists(modulesPath)) {
+                log(`Removing ${modulesPath}.`);
                 const error = await new Promise(resolve => rimraf(path.join(modulesPath), originalFs, resolve));
                 if (error) {
-                    log(`❌ Could not delete modules in ${roaming}`);
+                    log(`❌ Could not delete modules in ${discordConfigPath}.`);
                     log(`❌ ${error.message}`);
                     return error;
                 }
+                log("✅ Deletion successful.");
             }
-            log("✅ Deletion successful");
             progress.set(progress.value + progressPerLoop);
         }
         catch (err) {
-            log(`❌ Could not delete modules in ${roaming}`);
+            log(`❌ Could not delete modules in ${discordConfigPath}.`);
             log(`❌ ${err.message}`);
             return err;
         }
     }
 }
 
-async function showInstallNotice(config) {
-    const confirmation = await remote.dialog.showMessageBox(remote.BrowserWindow.getFocusedWindow(), {
-        type: "question",
-        title: "Reinstall BetterDiscord?",
-        message: "After repairing, you need to reinstall BetterDiscord. Would you like to do that now?",
-        noLink: true,
-        cancelId: 1,
-        buttons: ["Yes", "No"]
-    });
 
-    if (confirmation.response !== 0) return succeed();
+async function showInstallNotice(config, discordPaths) {
+    // Let the user know that they must launch Discord to download the "modules"
+    // folders that we deleted, otherwise we won't be able to inject since the
+    // files won't exist. If the user proceeds without reading, they'll get an error.
+    while (true) {
+        const confirmation = await remote.dialog.showMessageBox(remote.BrowserWindow.getFocusedWindow(), {
+            type: "question",
+            title: "Reinstall BetterDiscord?",
+            message: "After repairing, you need to reinstall BetterDiscord.\n\nBut first, you must manually launch all of your repaired Discord versions, so that they will download their modules again.\n\nWhen you are ready, return to this dialog and proceed with the BetterDiscord installation.\n\nAre you ready to install BetterDiscord now?",
```

Comment:
This is only relevant for Linux, but looks like this prompt will occur on Windows/Mac as well?

Reply:
It will show up and all platforms, and this is actually a potential issue that I spotted myself after submitting the PR, but I wasn't sure if Windows/Mac *also* needed to recreate the deleted folders. So I left it and awaited your review.

This will be a trivial fix, to change the code flow slightly, to only show the dialog if the user's actively selected install-folders are missing.

Basically:

- Check if config's install-folders all exist (would always be true on Windows/Mac, I think, but I will do a deeper look to make sure we don't delete any folders that BetterDiscord-installer tries to install into on Windows, or if such folders are automatically recreated by BD-installer on those platforms).
- If a critical installation path is missing, show the popup message.

This change will make it only appear on Linux and only when needed.